src/io/flutter/sdk/FlutterSdk.java

```diff
@@ -61,6 +70,59 @@ public static FlutterSdk getFlutterSdk(@NotNull final Project project) {
     return project.isDisposed() ? null : getFlutterSdkByDartSdk(DartPlugin.getDartSdk(project));
   }
 
+
+  /**
+   * Returns the Flutter SDK for a project that has a possibly broken "Dart SDK" project library.
+   * <p>
+   * (This can happen for a newly-cloned Flutter SDK where the Dart SDK is not cached yet.)
+   */
+  @Nullable
+  public static FlutterSdk getIncomplete(@NotNull final Project project) {
```

Comment:
Yep. I didn't rename the other one yet to keep the commit small.

New plan is to write a method to get the Flutter SDK of an unopened project. Then we can sync the SDK before project open begins.